src/common/AndroidUtils.ts

```diff
@@ -135,11 +135,10 @@ export class AndroidSDKUtils {
                     if (idx !== -1) {
                         reject(new Error('unsupported Java version.'));
                     }
-                    if (error.status === 127) {
+                    if (error.status && error.status === 127) {
                         reject(
                             new Error(
-                                'SDK Manager not found. Expected at ' +
-                                    AndroidSDKUtils.ANDROID_SDK_MANAGER_CMD
+                                `SDK Manager not found. Expected at ${AndroidSDKUtils.ANDROID_SDK_MANAGER_CMD}`
                             )
                         );
                     }
```

Comment:
Instead of `reject(error.message)` on line 145 perhaps we should just have `reject(error)` ?

Reply:
👍 Agreed. I think it's possible that this is the root of the issue we were seeing, specifically about getting `undefined` in the output. `androidSDKPrerequisitesCheck()`'s calling method, `isJava8Available()`, actually [expects an `Error` object back](https://github.com/forcedotcom/lwc-dev-mobile/blob/33cb21d0ac67b4ab0fe7255cf9562da96a588afa/src/common/AndroidEnvironmentSetup.ts#L97). In that case, `String.message` *is* going to be undefined.

Per our discussion offline, I also took the liberty of cleaning up the conditional flow. While a consumer of a `Promise` will only get back one rejection via its `catch`, that doesn't actually abort the upstream code flow. I assume the consumer simply drops the subsequent `reject()` calls, once fulfilled? At any rate, we shouldn't desire to have code continue to run, since it may have unintended side effects.

Reply:
Looks great